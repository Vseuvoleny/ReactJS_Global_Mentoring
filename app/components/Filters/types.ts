import { SortBy } from "../../types/types";

export interface IFilters {
    isFiltersOpen: boolean;
    setisFiltersOpen: Function;
    sortBy: SortBy;
}