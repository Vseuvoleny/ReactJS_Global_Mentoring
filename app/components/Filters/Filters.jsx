import React from "react";
import RedTriangle from "../../elements/Icon/Icons/RedTriangle/RedTriangle";
import classes from "./Filters.module.scss";

const Filters = ({
  isFiltersOpen,
  setisFiltersOpen,
  sortBy,
  setSearchParams,
}) => {
  return (
    <div className={classes.filter}>
      <span className={classes.filter__status}>sort by</span>
      <span
        className={classes.filter__options}
        onClick={() => setisFiltersOpen(!isFiltersOpen)}
      >
        {sortBy ?? "Release date"}
        <RedTriangle viewBox="0 0 10 7" width="10" height="7" fill="none" />
      </span>
      {isFiltersOpen && (
        <div className={classes.filter__expanded_list}>
          <div
            className={classes.filter__option}
            onClick={() => setSearchParams({ sortBy: "release_date" })}
          >
            Release date
          </div>
          <div
            className={classes.filter__option}
            onClick={() => setSearchParams({ sortBy: "genre" })}
          >
            Genre
          </div>
          <div
            className={classes.filter__option}
            onClick={() => setSearchParams({ sortBy: "rating" })}
          >
            Rating
          </div>
        </div>
      )}
    </div>
  );
};

export default Filters;
