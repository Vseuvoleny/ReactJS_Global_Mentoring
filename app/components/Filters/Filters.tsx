import React from "react";
import RedTriangle from "../../elements/Icon/Icons/RedTriangle/RedTriangle";
import Router from "next/router";
import classes from "./Filters.module.scss";

const Filters = ({ isFiltersOpen, setisFiltersOpen, sortBy }) => {
  return (
    <div className={classes.filter}>
      <span className={classes.filter__status}>sort by</span>
      <span
        className={classes.filter__options}
        onClick={() => setisFiltersOpen(!isFiltersOpen)}
      >
        {sortBy ?? "Release date"}
        <RedTriangle viewBox="0 0 10 7" width="10" height="7" fill="none" />
      </span>
      {isFiltersOpen && (
        <div className={classes.filter__expanded_list}>
          <div
            className={classes.filter__option}
            onClick={() =>
              Router.push({
                query: { ...Router.query, sortBy: "release_date" },
              })
            }
          >
            Release date
          </div>
          <div
            className={classes.filter__option}
            onClick={() =>
              Router.push({
                query: { ...Router.query, sortBy: "release_date" },
              })
            }
          >
            Genre
          </div>
          <div
            className={classes.filter__option}
            onClick={() =>
              Router.push({
                query: { ...Router.query, sortBy: "release_date" },
              })
            }
          >
            Rating
          </div>
        </div>
      )}
    </div>
  );
};

export default Filters;
