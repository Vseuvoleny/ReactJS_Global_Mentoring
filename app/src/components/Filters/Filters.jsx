import React from "react";
import RedTriangle from "../../Elements/Icon/Icons/RedTriangle/RedTriangle";

const Filters = ({ isFiltersOpen, setisFiltersOpen }) => {
  return (
    <div className="filter">
      <span className="filter__status">sort by</span>
      <span className="filter__options">
        release date{" "}
        <RedTriangle viewBox="0 0 10 7" width="10" height="7" fill="none" />
      </span>
    </div>
  );
};

export default Filters;
