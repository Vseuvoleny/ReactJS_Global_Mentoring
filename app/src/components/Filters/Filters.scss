.filter {
  position: relative;
  & > * {
    text-transform: uppercase;
    display: inline-block;
    line-height: 20px;
    font-size: 16px;
  }
  &__status {
    margin-right: 32px;
    opacity: 0.6;
  }
  &__options {
    font-weight: 500;
    svg {
      vertical-align: middle;
      margin-left: 12px;
    }
  }
  &__expanded-list {
    z-index: 9999;
    text-align: center;
    top: 35px;
    right: 0;
    position: absolute;
    width: 190px;
    background: rgba(35, 35, 35, 0.918051);
    box-shadow: 0px 0px 2px rgba(0, 0, 0, 0.1), 0px 2px 10px rgba(0, 0, 0, 0.1),
      0px 10px 20px rgba(0, 0, 0, 0.1), 0px 10px 50px rgba(0, 0, 0, 0.1);
    backdrop-filter: blur(27.1828px);
    & > * {
      padding: 6px 0;
      font-weight: 500;
      font-size: 16px;
      line-height: 20px;
      transition: 500ms;
      cursor: pointer;
      &:hover {
        background: #f65261;
      }
    }
  }
}
