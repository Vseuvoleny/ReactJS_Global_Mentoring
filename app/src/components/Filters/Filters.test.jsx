import React from "react";
import { render, screen } from "@testing-library/react";
import Filters from "./Filters";
import userEvent from "@testing-library/user-event";

describe("Filter component", () => {
  test("should render Filter component with opened filters ", () => {
    const { container, getByText, getAllByText } = render(
      <Filters isFiltersOpen={true} />
    );
    const option = getAllByText(/Release date/i)[1];

    expect(option).toBeInTheDocument();
    expect(getByText(/Genre/i)).toBeInTheDocument();
    expect(getByText(/Rating/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
