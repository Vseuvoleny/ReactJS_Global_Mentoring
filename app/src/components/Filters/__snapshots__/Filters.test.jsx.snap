// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component should render Filter component with opened filters  1`] = `
<div>
  <div
    class="filter"
  >
    <span
      class="filter__status"
    >
      sort by
    </span>
    <span
      class="filter__options"
    >
      Release date
      <svg
        fill="none"
        height="7"
        role="img"
        viewBox="0 0 10 7"
        width="10"
      >
        <path
          clip-rule="evenodd"
          d="M1.17339 0C0.319169 0 -0.141788 1.00184 0.413939 1.65057L4.23746 6.11398C4.63642 6.57971 5.35674 6.57992 5.75597 6.11442L9.58401 1.65101C10.1403 1.0024 9.67943 0 8.82494 0H1.17339Z"
          fill="#F65261"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="filter__expanded-list"
    >
      <div
        class="filter__option"
      >
        Release date
      </div>
      <div
        class="filter__option"
      >
        Genre
      </div>
      <div
        class="filter__option"
      >
        Rating
      </div>
    </div>
  </div>
</div>
`;
